src/bin/pgcopydb/copydb.c

```diff
@@ -787,8 +791,8 @@ copydb_init_table_specs(CopyTableDataSpec *tableSpecs,
 		.cfPaths = &(specs->cfPaths),
 		.pgPaths = &(specs->pgPaths),
 
-		.source_pguri = (char *) &(specs->source_pguri),
-		.target_pguri = (char *) &(specs->target_pguri),
+		.source_pguri = (char *) calloc(strlen(specs->source_pguri) + 1, sizeof(char)),
+		.target_pguri = (char *) calloc(strlen(specs->target_pguri) + 1, sizeof(char)),
```

Comment:
Again, either keep the same pointer the a memory area that's known to be kept around for the whole duration of the command, or use strdup.